app.py

```diff
@@ -151,100 +169,77 @@ async def delete_user(user_id):
-            pool.close()
-    if not token:
-        return jsonify({'error': 'Invalid refresh token.'}), 401
+    refresh_token = request.json["refresh_token"]
     try:
-        jwt.decode(refresh_token, app.config['SECRET_KEY'], algorithms=['HS256'])
+        token_payload = jwt.decode(refresh_token, app.config["SECRET_KEY"], algorithms=["HS256"])
     except jwt.ExpiredSignatureError:
-        return jsonify({'error': 'Refresh token has expired.'}), 401
-    access_token = jwt.encode({'user_id': token[0], 'exp': datetime.datetime.utcnow() + datetime.timedelta(minutes=60)},
-                              app.config['SECRET_KEY'],
-                              algorithm='HS256')
-    refresh_token = jwt.encode({'user_id': token[0], 'exp': datetime.datetime.utcnow() + datetime.timedelta(days=7)},
-                               app.config['SECRET_KEY'], algorithm='HS256')
-    access_expire = datetime.datetime.utcnow() + datetime.timedelta(minutes=60)
-    refresh_expire = datetime.datetime.utcnow() + datetime.timedelta(days=7)
-    pool = await create_pool()
-    async with pool.acquire() as conn:
-        async with conn.cursor() as cur:
-            await cur.execute('INSERT INTO refresh_tokens (user_id, refresh_token) VALUES (%s, %s)',
-                              (token[0], refresh_token))
-            await cur.close()
-            pool.close()
-    return jsonify({'access_token': access_token, 'refresh_token': refresh_token, 'access_expire': access_expire,
-                    'refresh_expire': refresh_expire}), 200
+        return jsonify({"error": "Refresh token has expired."}), 401
+    except jwt.InvalidTokenError:
+        return jsonify({"error": "Invalid refresh token."}), 401
+    return create_and_return_token(token_payload["user_id"])
+
+
+def create_and_return_token(user_id):
+    access_expire = int(time.time()) + 3600
+    refresh_expire = int(time.time()) + 604800
+    access_token = jwt.encode({"user_id": user_id, "exp": access_expire},
+                              app.config["SECRET_KEY"],
+                              algorithm="HS256")
+    refresh_token = jwt.encode({"user_id": user_id, "exp": refresh_expire},
+                               app.config["SECRET_KEY"], algorithm="HS256")
+    return jsonify({"access_token": access_token, "refresh_token": refresh_token,
+                    "access_expire": datetime.fromtimestamp(access_expire),
+                    "refresh_expire": datetime.fromtimestamp(refresh_expire)}), 200
 
 
 # Add pin
-@app.post('/pins')
+@app.post("/pins")
 @token_required
 async def add_pin(current_user):
-    title = request.form['title']
-    body = request.form['body']
-    f = request.files.get('image')
-    if title == '' or body == '' or f is None:
-        return jsonify({'error': 'Field is empty'}), 400
-    filename = secure_filename(f.filename)
-    f.save(filename)
+    required_fields = {"title": request.form.get("title"), "body": request.form.get("body"),
```

Comment:
You may use a simple approach like bellow.

`required_fields = ["title", "body", "image"]`
 `for _filed in required_fields:`
       `if not _filed in request.form:`
             `return jsonify({"error": f"{_filed} is required."}), 400`